lib/nexthop.c

```diff
@@ -1158,3 +1158,331 @@ bool nexthop_is_ifindex_type(const struct nexthop *nh)
 		return true;
 	return false;
 }
+
```

Comment:
Part of the reason I've never moved to having a vty function and a json function is that you end up with code that is in two functions and an addition to the data structure being dumped is easy to forget the json aspect of it ( or vice versa ).  It always sucked but I felt the in your face aspect was warranted.  What advantage are you looking for with this change?  How do we ensure that a new developer is going to remember to hit both sides of the equation?